libs/api/utils/src/lib/api-utils.ts

```diff
@@ -1,13 +1,37 @@
 import { prisma } from '@nmit-coursition/db'
 import { isDateBeforeNow } from '@nmit-coursition/utils'
+import { Prisma } from '@prisma/client'
 import { parseApiKey } from '../api'
 import type { ApiErrorCode } from '../errorList'
+import type { ApiKeyReportUsageRequest, ApiUsageReport, ApiUsageRequest } from '../typescript'
+
+let API_KEY_TO_ID_CACHE: { [key: string]: bigint } = {}
+
+let API_KEY_CONTEXT: string | undefined
 
 export function reportUsage(apiKey: string, duration: number, type: 'video' | 'document' | 'web') {
   // eslint-disable-next-line no-console -- will be replaced with real usage reporting
   console.log(`API Key ${apiKey} used ${duration} on ${type}.`)
 }
 
+export async function reportSpend({ apiKey, operationClass, spend, metaData }: ApiKeyReportUsageRequest) {
+  const key = apiKey || API_KEY_CONTEXT
+  if (!key) throw new Error(`API key context has not been set. Please define apiKey.`)
+
+  // eslint-disable-next-line no-console debug info
+  console.log(`🚀 Spend report (${operationClass || 'A'}): ${spend || 1}`)
+
+  await prisma.core__api_key_usage.create({
+    data: {
+      key_id: await resolveApiKeyIdByKey(key),
+      operation_class: operationClass || 'A',
```

Comment:
**nitpick:** Better way to handle this is default parameter in function definition, e.g. `operationClass='A'`.